var/spack/repos/builtin/packages/llvm-amdgpu/package.py

```diff
@@ -59,6 +60,7 @@ class LlvmAmdgpu(CMakePackage):
 
     # Add device libs sources so they can be an external LLVM project
     for d_version, d_shasum in [
+        ('4.3.0',  '055a67e63da6491c84cd45865500043553fb33c44d538313dd87040a6f3826f2'),
```

Comment:
need to add libcxx; libcxxabi to the llvm_projects below for generating amdclang binary.
also the following flags are required 
   LIBCXX_ENABLE_SHARED=OFF 
   LIBCXX_ENABLE_STATIC=ON 
   LIBCXX_INSTALL_LIBRARY=OFF 
   LIBCXX_INSTALL_HEADERS=OFF 
   LIBCXXABI_ENABLE_SHARED=OFF 
   LIBCXXABI_ENABLE_STATIC=ON 
   LIBCXXABI_INSTALL_STATIC_LIBRARY=OFF

Reply:
is this PR work in progress?

Reply:
I think so, but for now as part of this PR i think we can the following -
Starting rocm-4.3.0 . Enable libcxx, libcxxabi  inside llvm-projects so that we have new binaries amdclang/amdclang++/amdflang getting generated.

In a new PR  we can have the following -
inside the install prefix of llvm-amdgpu , 
rename /bin into /llvm  subdirectory. 
create /bin and Inside /bin - create symlinks for amdclang/amdclang++/amdflang  that are inside /llvm 

A new recipe for close source binaries and inside the llvm-amd-gpu , have a new variant which is optional ,that can download the close source binaries and  inside the llvm-amdgpu install prefix create a new /alt sub directory and create symlinks to new closed source binaries.
Based on this we need to make changes elsewhere too for the other rocm recipes.

Reply:
Appended cmake variables, let get the new changes after the closed source llvm compiler recipe is added